src/test/java/com/rocketden/main/UserIntegrationTests.java

```diff
@@ -0,0 +1,134 @@
+package com.rocketden.main;
+
+import static org.junit.jupiter.api.Assertions.*;
+
+import java.lang.reflect.Type;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.concurrent.CountDownLatch;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.atomic.AtomicReference;
+
+import com.rocketden.main.controller.v1.BaseRestController;
+import com.rocketden.main.model.User;
+
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.boot.web.server.LocalServerPort;
+import org.springframework.messaging.converter.MappingJackson2MessageConverter;
+import org.springframework.messaging.simp.stomp.StompCommand;
+import org.springframework.messaging.simp.stomp.StompFrameHandler;
+import org.springframework.messaging.simp.stomp.StompHeaders;
+import org.springframework.messaging.simp.stomp.StompSession;
+import org.springframework.messaging.simp.stomp.StompSessionHandler;
+import org.springframework.messaging.simp.stomp.StompSessionHandlerAdapter;
+import org.springframework.web.socket.WebSocketHttpHeaders;
+import org.springframework.web.socket.client.standard.StandardWebSocketClient;
+import org.springframework.web.socket.messaging.WebSocketStompClient;
+import org.springframework.web.socket.sockjs.client.SockJsClient;
+import org.springframework.web.socket.sockjs.client.Transport;
+import org.springframework.web.socket.sockjs.client.WebSocketTransport;
+
+@SpringBootTest(webEnvironment = SpringBootTest.WebEnvironment.RANDOM_PORT)
+public class UserIntegrationTests {
+
+    @LocalServerPort
+    private int port;
+
+    private SockJsClient sockJsClient;
+
+    private WebSocketStompClient stompClient;
+
+    private final WebSocketHttpHeaders headers = new WebSocketHttpHeaders();
+
+    @BeforeEach
+    public void setup() {
+        // Create variables to model message transport, the SockJS client, and the STOMP client.
+        List<Transport> transports = new ArrayList<>();
+        transports.add(new WebSocketTransport(new StandardWebSocketClient()));
+        this.sockJsClient = new SockJsClient(transports);
+        this.stompClient = new WebSocketStompClient(sockJsClient);
+        this.stompClient.setMessageConverter(new MappingJackson2MessageConverter());
+    }
+
+    @Test
+    public void getUser() throws Exception {
```

Comment:
It is possible to extract some of these overridden classes into separate methods/utility files (bonus if they can then be reused for other socket tests)? If not, could you add a summary comment at the top explaining what this test does?

Reply:
Right now, I'm hesitant to separate those overridden classes into separate files because this is the only test that uses them. If it's OK, I'm thinking we can move these methods into utility files when multiple tests use them; in the meantime however, I added a summary comment at the top explaining what the test does 🙂